src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs

```diff
@@ -47,23 +62,102 @@ public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options)
             }
         }
 
+        private static ConcurrentDictionary<string, IConsoleLogFormatter> SetFormatters(IEnumerable<IConsoleLogFormatter> fromSetup = null)
+        {
+            IEnumerable<IConsoleLogFormatter> formatters = fromSetup;
+            if (formatters == null || formatters.ToList().Count == 0)
+            {
+                var defaultMonitor = new FormatterOptionsMonitor<DefaultConsoleLogFormatterOptions>(new DefaultConsoleLogFormatterOptions());
+                var systemdMonitor = new FormatterOptionsMonitor<SystemdConsoleLogFormatterOptions>(new SystemdConsoleLogFormatterOptions());
+                formatters = new List<IConsoleLogFormatter>()
+                {
+                    new DefaultConsoleLogFormatter(defaultMonitor),
+                    new SystemdConsoleLogFormatter(systemdMonitor)
+                };
+            }
+            return new ConcurrentDictionary<string, IConsoleLogFormatter>(formatters.ToDictionary(f => f.Name));
```

Comment:
>ormatters.ToDictionary(f => f.Name) [](start = 75, length = 35)

we can avoid the extra allocation of the intermediate dictionary if we just enum the llist and add it directly to the ConcurrentDictionary